src/DeviceStatusPage/index.tsx

```diff
@@ -190,6 +203,20 @@ export default (): React.ReactElement => {
       });
   }, [astarte.client, deviceId, dismissModal, devicePageAlersController]);
 
+  const handleDeleteDevice = useCallback(() => {
+    astarte.client
+      .deleteDevice(deviceId)
+      .then(() => {
+        setActiveModal({ kind: 'delete_device_modal', isDeletingDevice: true });
+        deviceFetcher.refresh();
+        navigate('/devices');
```

Comment:
Since we're navigating to a different page, there's no need to refresh the device query.